File: backend/src/controller/Order.js
const { cart } = require("../models/Cart");
const { order } = require("../models/Order");
const { product } = require("../models/Product");

const createOrder = async (req, res, next) => {
  try {
    const user = req.user;
    const { productOrder } = req.body;
    if (!productOrder) {
      return res.status(400).json({
        message: "productOrder is required",
      });
    }
    console.log(productOrder);

    var orderItem = [];

    productOrder.forEach((item) => {
      orderItem.push({ product: item.product._id, size: item.size });
    });

    let totalPrice = 0;
    for (let item of productOrder) {
      const foundProduct = await product.findById(item.product._id);

      if (!foundProduct) {
        return res
          .status(404)
          .json({ message: `Product with id ${item.product._id} not found` });
      }

      totalPrice += foundProduct.price;
    }

    await cart.deleteMany({ user: user._id });

    const newOrder = new order({
      user: user._id,
      productOrder: orderItem,
      totalPrice,
    });

    await newOrder.save();

    return res.status(201).json({
      message: "Order created successfully",
      // result: newOrder
    });
  } catch (error) {
    console.error("Error creating order:", error);
    return res.status(500).json({ message: "Internal server error" });
  }
};
const getAllOrderForUser = async (req, res) => {
  try {
    const orders = await order
      .find()
      .populate({
        path: "productOrder",
        populate: {
          path: "product",
          model: "Product",
          select: "_id name price image",
        },
      })
      .sort({ updatedAt: -1 });
    return res.status(200).json(orders);
  } catch (error) {
    console.error("Error creating order:", error);
    return res.status(500).json({ message: "Internal server error" });
  }
};

module.exports = {
  createOrder,
  getAllOrderForUser,
};

File: fontend/lib/models/order.dart
import 'package:fontend/models/order_item.dart';

class Order {
  String? id;
  String? user;
  List<OrderItem>? productOrder;
  int? totalPrice;
  String? status;
  String? createdAt;
  String? updatedAt;

  Order({
    this.id,
    this.user,
    this.productOrder,
    this.totalPrice,
    this.status,
    this.createdAt,
    this.updatedAt,
  });

  Order.fromJson(Map<String, dynamic> json) {
    id = json["_id"];
    user = json["user"];
    if (json["productOrder"] != null) {
      List<dynamic> list = json["productOrder"];
      productOrder = list.map((item) => OrderItem.fromJson(item)).toList();
    } else {
      productOrder = [];
    }
    totalPrice = json["totalPrice"];
    status = json["status"];
    createdAt = json["createdAt"];
    updatedAt = json["updatedAt"];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['_id'] = id;
    data['user'] = user;
    data['productOrder'] = productOrder;
    data['totalPrice'] = totalPrice;
    data['status'] = status;
    return data;
  }
}
